### ISSUE TITLE
Add Google OAuth register and login to OSP

### ISSUE THREAD

**Issue by bismitaguha** [anitab-org/open-source-programs-backend#30]
## Description

As a developer,
I need to implement Google sign on,
so that I can allow Google users to login using their Google account.

For being able to use the application, the user has to create a new account and confirm the email just to login. It's a long process. So by having Google OAuth we can make the process shorter for users who have a Google account.

### Update [Required]
- [ ] Implement Google OAuth login and registration
- [ ] Write unit tests for the updates ^^

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
8 hours

**Comment by saideepesh000** [comment]
@bismitaguha, are we going to update the front-end of the app simultaneously, or else after completing this in the backend, we add this to the front-end?

**Comment by bismitaguha** [comment]
> @bismitaguha, are we going to update the front-end of the app simultaneously, or else after completing this in the backend, we add this to the front-end?

Yes the issue needs to be addressed in the frontend too, can be picked up by anybody!

**Comment by chinmaym07** [comment]
Hey can I work on this issue .?

**Comment by Aaishpra** [comment]
@isabelcosta @vj-codes I would like to work here.

**Comment by vj-codes** [comment]
@chinmaym07 hey are you still interested to work on this issue ? If there's no response, it will be assigned to the contributor who claimed it next

**Comment by chinmaym07** [comment]
> @chinmaym07 hey are you still interested to work on this issue ? If there's no response, it will be assigned to the contributor who claimed it next

If it's available ,Yes I am interested to work on it ... @vj-codes

**Comment by isabelcosta** [comment]
@chinmaym07 its assigned to you! Happy coding!

**Comment by isabelcosta** [comment]
How's your issue work going @chinmaym07 ? Do you need any help or support?

**Comment by chinmaym07** [comment]
Yes I am working on this issue. But I am facing this issue.
After login ,I am regularly keep getting this error whenever I try to update the details of the user , it shows me this error.
![Screenshot from 2021-01-27 12-41-47](https://user-images.githubusercontent.com/44670961/105955999-1f86ec00-609d-11eb-8c98-860eff9f0f04.png)

Can you @isabelcosta please help me with that.

**Comment by chinmaym07** [comment]
@isabelcosta I have got the file that contains this info .
https://anitab-org.zulipchat.com/#narrow/stream/237907-open-source-progs/topic/Questions/near/224148645
Thanks to sankalp.! 😁 
But still the above error is coming

**Comment by isabelcosta** [comment]
Did you manage to get help on Zulip? @chinmaym07

**Comment by chinmaym07** [comment]
Yes @isabelcosta .I have implemented google authentication . Only Github one is left ..which I am working on .
I think above error is solved by anitab-org/open-source-programs-web#51.

**Comment by isabelcosta** [comment]
@chinmaym07 do you wish to deliver in one PR the Google OAuth, and on another issue the GitHub OAuth? I am okay with that, and I would even recommend it. Unless you already started that work and it creates more struggle for you

**Comment by chinmaym07** [comment]
Yes ..!! @isabelcosta

**Comment by codesankalp** [comment]
@chinmaym07 Any updates on this?
You can make a PR of Google OAuth if Github one is still remaining.

**Comment by isabelcosta** [comment]
yap, @chinmaym07 it is actually better if you just do one of the OAuth, and someone else can do it on another PR :) So that the PR is focused on one specific "little" task. Can you please choose one OAuth service, so we can update the issue.

**Comment by isabelcosta** [comment]
I will update this to be about Google OAuth, and leave the GitHub to another.
@codesankalp ~could you please create an issue for implementing GitHub OAuth?~
I can create real quick, nevermind

**Comment by chinmaym07** [comment]
@codesankalp , @isabelcosta  sorry for taking so long to implement this . As my exams are going on. I will submit the pr asap. I have implemented the google one just testing it using the frontend. It's working using the frontend .
@codesankalp I think this issue only needs the Google auth api endpoint to be created in order to call it using the frontend. please correct me if there is any additional thing .

**Comment by isabelcosta** [comment]
No problem @chinmaym07 I understand a lot of contributors in the community now are going through exams. You can update us from time to time. slow progress is still progress ;)
I think reducing the scope for google login was the right option here :)

**Comment by chinmaym07** [comment]
Update : Completed the backend implementation & tested using the frontend .
@codesankalp  @isabelcosta Can I also open the frontend issue for this ? 
Will make the PR by today 😀

**Comment by codesankalp** [comment]
Great @chinmaym07 :tada: 
You can open the frontend issue for this.
Looking forward to seeing your PR.
